openmc/material.py

```diff
@@ -914,6 +915,97 @@ def to_xml_element(self, cross_sections=None):
 
         return element
 
+    @classmethod
+    def mix_materials(cls, materials, fracs, percent_type='ao', name=None):
+        """Mix materials together based on atom, weight, or volume fractions
+
+        Parameters
+        ----------
+        materials : Iterable of openmc.Material
+            Materials to combine
+        fracs : Iterable of float
+            Fractions of each material to be combined
+        percent_type : str
+            Type of percentage, must be one of 'ao', 'wo', or 'vo', to signify atom
+            percent (molar percent), weight percent, or volume percent, 
+            optional. Defaults to 'ao'
+        name : str
+            The name for the new material, optional. Defaults to concatenated
+            names of input materials with percentages indicated inside
+            parentheses.
+
+        Returns
+        -------
+        new_mat : openmc.Material instance for the new mixture
+
+        """
+
+        cv.check_type('materials', materials, Iterable, Material)
+        cv.check_type('fracs', fracs, Iterable, Real)
+        cv.check_value('percent type', percent_type, {'ao', 'wo', 'vo'})
+
+        fracs = np.asarray(fracs)
+        void_frac = 1. - np.sum(fracs)
+
+        # Warn that fractions don't add to 1, set remainder to void, or raise
+        # an error if percent_type isn't 'vo'
+        if not np.isclose(void_frac, 0.):
+            if percent_type in ['ao', 'wo']:
+                msg = 'A non-zero void fraction is not acceptable for \
+                    percent_type: {}'.format(percent_type)
```

Comment:
Same issue here regarding whitespace in the middle of the string (see comment at line 979)